### PR TITLE
SPAdes: add python dependency

### PR DESCRIPTION
**Pull request by jbedo**
###### Motivation for this change

spades.py requires python.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/spades/default.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
-  buildInputs = [ zlib bzip2 ];
+  buildInputs = [ zlib bzip2 python ];
```
Comment (jtojnar) [inline pkgs/applications/science/biology/spades/default.nix:14]: Please use either `python2` or `python3`.

**Review by jtojnar — CHANGES_REQUESTED** [review]
The commit message tag needs to match the attribute name `SPAdes`.

**Review by bzizou — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Build in #87494 fine without. Closing.

**Comment by jbedo** [comment]
Problem was in execution not build, python is a runtime dependency.

**Comment by jbedo** [comment]
Confirmed, patch still required:

```
~/g/nixpkgs on 8256fc2da56 ⨯ nix shell .#SPAdes -c spades.py
/usr/bin/env: ‘python’: No such file or directory    
```

**Comment by jbedo** [comment]
SPAdes is unusable without this patch, is there anything further I need to do to get it merged? Might be worth considering backporting to 20.09 since it's also broken there.

**Review by bzizou — APPROVED** [review]
I agree that the python dependency allows spades.py to run in a pure environment without explicitly loading python

**Comment by SuperSandro2000** [comment]
I read on their GitHub that they support python3 https://github.com/ablab/spades#installation. Can we please use python3 as we are currently remove python2 from as many packages as possible.

**Comment by jbedo** [comment]
Changed to python 3, rebased against current master, and verified functional.

**Review by bzizou — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 89778` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>SPAdes</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 89778` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>SPAdes</li>
  </ul>
</details>

**Pull request merged**